national_baf_cd_population_request_09_28_2022/main.ipynb

```diff
@@ -0,0 +1,1807 @@
+       "022            True\n",
+       "023            True\n",
+       "024            True\n",
+       "025            True\n",
+       "026            True\n",
+       "027            True\n",
+       "028            True\n",
+       "029            True\n",
+       "030            True\n",
+       "031            True\n",
+       "032            True\n",
+       "033            True\n",
+       "034            True\n",
+       "035            True\n",
+       "036            True\n",
+       "037            True\n",
+       "038            True\n",
+       "039            True\n",
+       "040            True\n",
+       "041            True\n",
+       "042            True\n",
+       "044            True\n",
+       "045            True\n",
+       "046            True\n",
+       "047            True\n",
+       "048            True\n",
+       "049            True\n",
+       "050            True\n",
+       "051            True\n",
+       "053            True\n",
+       "054            True\n",
+       "055            True\n",
+       "056            True"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def check_state_totals():\n",
+    "    natpl['statefips'] = natpl['GEOID20'].str.slice(stop=3)\n",
+    "    natpl['P0010001'] = natpl['P0010001'].astype(int)\n",
+    "    pl_gpby = natpl.groupby(['statefips']).sum()\n",
+    "    \n",
+    "    baf_pl['statefips'] = baf_pl['GEOID20'].str.slice(stop=3)\n",
+    "    baf_pl['P0010001'] = baf_pl['P0010001'].astype(int)\n",
+    "    baf_pl_gpby = baf_pl.groupby(['statefips']).sum()\n",
+    "    \n",
+    "    return pl_gpby == baf_pl_gpby\n",
+    "    \n",
+    "check_state_totals()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "0522dea9-49ec-4a50-8601-1da6c1efaa44",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "<ipython-input-10-eb491769f8b9>:2: SettingWithCopyWarning: \n",
+      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
+      "Try using .loc[row_indexer,col_indexer] = value instead\n",
+      "\n",
+      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
+      "  baf_pl['CD_ID'] = baf_pl['STATEAB'].astype(str) + '-' + baf_pl['CONG'].astype(str).str.upper().str.zfill(3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "baf_pl = baf_pl[(~baf_pl['STATEAB'].isna())]#|(baf_pl['P0010001']!=0)]\n",
+    "baf_pl['CD_ID'] = baf_pl['STATEAB'].astype(str) + '-' + baf_pl['CONG'].astype(str).str.upper().str.zfill(3)\n",
+    "\n",
+    "cd = gp.read_file(f'zip+s3://data.redistrictingdatahub.org/web_ready_stage/NATIONAL/national_cong_2022.zip')\n",
+    "cd['CD_ID'] = cd['STATE'].astype(str) + '-' + cd['DISTRICT'].astype(str).str.upper().str.zfill(3)\n",
```

Comment:
One thing that could be "changeable" would be this zfill. 

Because no state has more than 99 districts, I think they all should have been zfilled to 2 and not 3.

Reply:
This was zfilled to 3 because that is how it appears in the national Congressional file. So when I changed zfill to 2 it created more states that did not join, so for now I left the zfill at 3, but MS district assignment still only appears as 2. If the updated file only has two digits change the zfill to that when you modify @peter-redistricting-data-hub . Noted this in the code as well with where to insert updated s3 link (but hide before posting). Thanks!